backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/split/ShipmentScheduleSplitRepository.java

```diff
@@ -25,6 +27,7 @@
 public class ShipmentScheduleSplitRepository
 {
 	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IInOutBL inOutBL = Services.get(IInOutBL.class);
```

Comment:
repositories shall not use other services that use other repositories.

So, pls move the inOutBL to ShipmentScheduleSplitService and apply the logic there.
Keep the repository stupid simple, focused retrieval/searching and other CRUD operations.
`service` is the place where you can use multiple services/repostitories and make them interoperate.

Reply:
done